submission_worker.py

```diff
@@ -0,0 +1,300 @@
+        with open(download_location, 'w') as f:
+            f.write(response.content)
+        # extract zip file
+        zip_ref = zipfile.ZipFile(download_location, 'r')
+        zip_ref.extractall(extract_location)
+        zip_ref.close()
+        # delete zip file
+        try:
+            os.remove(download_location)
+        except Exception as e:
+            print 'Failed to remove zip file {}, error {}'.format(download_location, e)
+
+
+def create_dir(directory):
+    '''
+        Creates a directory if it does not exists
+    '''
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+def extract_challenge_data(challenge, phases):
+    '''
+        * Expects a challenge object and an array of phase object
+        * Extracts `evaluation_script` for challenge and `annotation_file` for each phase
+
+    '''
+
+    challenge_data_directory = CHALLENGE_DATA_DIR.format(challenge_id=challenge.id)
+    evaluation_script_url = challenge.evaluation_script
+    # create challenge directory
+    create_dir(challenge_data_directory)
+
+    challenge_zip_file = join(challenge_data_directory, 'challenge_{}.zip'.format(challenge.id))
+    download_and_extract_file(evaluation_script_url, challenge_zip_file, challenge_data_directory)
+
+    phase_data_base_directory = join(challenge_data_directory, 'phase_data')
+
+    for phase in phases:
+        phase_data_directory = join(phase_data_base_directory, str(phase.id))
+        # create phase directory
+        create_dir(phase_data_directory)
+        phase_zip_file = join(phase_data_directory, 'phase_{}.zip'.format(phase.id))
+        annotation_file_url = phase.test_annotation
+        download_and_extract_file(annotation_file_url, phase_zip_file, phase_data_directory)
+
+
+def load_active_challenges():
+    '''
+         * Fetches active challenges and corresponding active phases for it.
+    '''
+    q_params = {}
+    q_params['start_date__lt'] = timezone.now()
+    q_params['end_date__gt'] = timezone.now()
+
+    # make sure that the challenge base directory exists
+    create_dir(BASE_CHALLENGE_DATA_DIR)
+
+    active_challenges = Challenge.objects.filter(**q_params)
+
+    for challenge in active_challenges:
+        phases = challenge.challengephase_set.all()
+        extract_challenge_data(challenge, phases)
+
+
+def extract_submission_data(submission_id):
+    '''
+        * Expects submission id and extracts input file for it.
+    '''
+
+    try:
+        submission = Submission.objects.get(id=submission_id)
+    except Submission.DoesNotExist:
+        print 'Submission {} does not exist'.format(submission_id)
+
+    submission_input_file = submission.input_file
+    submission_data_directory = SUBMISSION_DATA_DIR.format(submission_id=submission.id)
+    # create submission directory
+    create_dir(submission_data_directory)
+
+    submission_zip_file = join(submission_data_directory, 'submission_{}.zip'.format(submission.id))
```

Comment:
same doubt as above. Are the submission files just one prediction file or it is a zip?